{{cookiecutter.project_name}}/include/itkNormalDistributionImageSource.hxx

```diff
@@ -63,7 +63,7 @@ NormalDistributionImageSource< TImage >
 
   typedef ImageScanlineIterator< ImageType > IteratorType;
   IteratorType it( output, outputRegion );
-  ProgressReporter progress( this, threadId, numberOfLinesToProcess );
+  ProgressReporter progress( this, 0, numberOfLinesToProcess );
```

Comment:
`ProgressReporter` should be completely removed, as progress is handled by MulitThreaders for DynamicThreadedGenerateData.

Reply:
I'd dare to say that tests rather use the `itk::SimpleFilterWatcher` (formerly `FilterWatcher`) since [this gerrit topic](http://review.source.kitware.com/#/c/23415/) was merged. I believe [this gerrit tiopic](http://review.source.kitware.com/#/c/23382/) may be related also. @dzenanz correct me if I'm wrong.

Progress on this for the remotes is being commented [here](http://review.source.kitware.com/#/c/23434/). Basically, it's been done for them :smile:. I trusted the new framework and did not check at some random test whether the progress was being reported as expected with the new framework :man_shrugging: :woman_shrugging:

Reply:
That is invoked [here](https://github.com/InsightSoftwareConsortium/ITK/blob/d2839085ef9378c64ce698db275cdb024e13fa8e/Modules/Core/Common/src/itkMultiThreaderBase.cxx#L453-L478).

Reply:
Progress is reported per work unit, but only by the invoking thread. So if you only have 4 work units, you will mostly get just 2-3 updates on a 4-core processor. If you use TBBMultiThreader, you get more updates. And once the thread/work unit split is done, there should be more updates for all multi-threaders. You could look at this [test](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/test/itkMultiThreaderParallelizeArrayTest.cxx) to see how progress reporting looks like, and how skewed it is by when the main thread is executing.